home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/RSQLNewsIT.java

```diff
@@ -0,0 +1,277 @@
+        sort = "source,asc";
+        List<String> sourceList = new ArrayList<>();
+
+        List<ReadNews> listNews = newsApi
+                .getAllNews(pageNumber,
+                        pageSize,
+                        sort,
+                        filter,
+                        id,
+                        title,
+                        text,
+                        source);
+        listNews.forEach(readUser -> sourceList.add(readUser.getSource()));
+        String[] actualSource = new String[sourceList.size()];
+        sourceList.toArray(actualSource);
+        Collections.sort(sourceList);
+        String[] expectedSource = new String[sourceList.size()];
+        sourceList.toArray(expectedSource);
+
+        assertArrayEquals(expectedSource, actualSource);
+    }
+
+    @Test
+    void getAllUsersSortAndSearchTest() throws ApiException {
+        List<String> sourceList = new ArrayList<>();
+        sort = "source,asc";
+        filter = new StringBuilder()
+                .append("id=bt=('")
+                .append(newsList.get(1).getId())
+                .append("',")
+                .append("'")
+                .append(newsList.get(newsList.size() - 1).getId())
+                .append("')").toString();
+
+        List<ReadNews> listNews = newsApi
+                .getAllNews(pageNumber,
+                        pageSize,
+                        sort,
+                        filter,
+                        id,
+                        title,
+                        text,
+                        source);
+        listNews.forEach(readUser -> sourceList.add(readUser.getSource()));
+        String[] actualSource = new String[sourceList.size()];
+        sourceList.toArray(actualSource);
+        Collections.sort(sourceList);
+        String[] expectedSource = new String[sourceList.size()];
+        sourceList.toArray(expectedSource);
+
+        assertEquals(listNews.size(), 4);
+        assertArrayEquals(expectedSource, actualSource);
+    }
+
+    @Test
+    void wrongArgumentTest() {
+        text = " ";
+
+        assertThatExceptionOfType(ApiException.class)
+                .isThrownBy(() -> newsApi.getAllNews(pageNumber,
+                        pageSize,
+                        sort,
+                        filter,
+                        id,
+                        title,
+                        text,
+                        source))
+                .withMessage(new StringBuilder()
+                        .append("{").append("\"responseCode\"")
+                        .append(":400,")
+                        .append("\"errorMessage\"")
+                        .append(":")
+                        .append("\"Illegal argument for request.\"")
+                        .append("}").toString());
+
+    }
+
+
+    @AfterEach
+    void tearDown() throws ApiException {
+        for (ReadNews ru : newsList) {
+            newsApi.deleteNews(ru.getId());
```

Comment:
I personally think that the data we created during test running could be useful after test completed. We can access it through postman and verify additionally when we troubleshoot some scenarios.

Please explain the purpose of deleting this data.